File: 2024/day.cs
namespace aoc2024;

public interface IDay {
    public (object, object) Run();
}

public interface Day<I1, I2, O1, O2> : IDay {
    public (I1, I2) Prepare(string in_string);
    public O1 Part1(I1 in_data);
    public O2 Part2(I2 in_data);
}

File: 2025/src/day.kt
typealias DayReturn = Pair<Any, Any>

public interface IDay {
	fun run(): DayReturn
}

public interface Day<I1, I2, O1, O2> : IDay {
	fun prepare(inString: String): Pair<I1, I2>
	fun part1(inData: I1): O1
	fun part2(inData: I2): O2
}